components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/model/AuthType.java

```diff
@@ -73,18 +116,41 @@ public AuthenticationType getType() {
         return type;
     }
 
-    public Map<String, Object> getProperties() {
+    public void setProperties(List<AuthProperty> properties) {
+
+        this.properties = properties;
+    }
+
+    public List<AuthProperty> getProperties() {
 
         return properties;
     }
 
+    public List<AuthProperty> getPropertiesWithDecryptedValues(String actionId) throws SecretManagementException {
+
+        if (properties != null) {
+
+            return secretProcessor.decryptAssociatedSecrets(properties, actionId, type.name());
+        }
+        return null;
+    }
+
+    public List<AuthProperty> getPropertiesWithSecretReferences(String actionId) throws SecretManagementException {
+
+        if (properties != null) {
+
+            return secretProcessor.getPropertiesWithSecretReferences(properties, actionId, type.name());
+        }
+        return null;
```

Comment:
Will this be needed ?

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5795/commits/4c40292ff2e65531731f8e68897c74215dae5279